tests/test_backend.py

```diff
@@ -3010,6 +3010,13 @@ def test_hashtable_lookup(self):
         self._run_test_case([lookup_results.name], {_INPUT: query})
         os.remove(filnm)
 
+    @check_opset_min_version(11)
+    def test_MatrixDiagPart(self):
+        input_val = np.array([[[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15], [16, 17, 18, 19, 20]]]).astype(
+            np.int64)
```

Comment:
Is it possible to test out higher dimensional tensors as well? The current tensor seems 2Dimensional ( 4x4).

Maybe we can try random values in shapes of  [ (n1, n2, n3),   (n1, n2, n3, n4),  (n1, n2, n3, n4, n5)] in a loop.

Also in the above it might be good to to ensure that  n5 is **smaller** than n4 above -- since there's a slight variation in the output shape based on which of the last 2 dimensions is larger.


```
https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/matrix-diag-part

Output: The extracted diagonal(s) having shape diagonal.shape = input.shape[:-2] + [min(input.shape[-2:])].
```

Reply:
Good point, let's add them.